docs/essentials/resources.md

```diff
@@ -0,0 +1,66 @@
+
+## Simple Usage
+
+The most straight-forward way to use a managed resource is with the `use` method:
+
+```scala mdoc:silent
+import zio._
+import zio.blocking.Blocking
+import zio.nio.core.channels._
+import zio.nio.core.file.Path
+import java.io.IOException
+
+def useChannel(f: FileChannel): ZIO[Blocking, IOException, Unit] = ???
+
+val effect: ZIO[Blocking, IOException, Unit] = FileChannel.open(Path("foo.txt"))
+  .use { fileChannel =>
+    // fileChannel is only valid in this lexical scope
+    useChannel(fileChannel)
+  }
+```
+
+In the above example, the `FileChannel` will be opened and then provided to the function passed to `use`. The channel will always be closed when the `use` function completes, regardless of whether the operation succeeds, fails, dies or is interrupted. As long as the channel is only used within the function passed to `use`, then we're guaranteed not to have leaks.
+
+## Flexible Resource Scoping
+
+Sometimes there are situations where `ZManaged#use` is too limiting, because the resource lifecycle needs to extend beyond a lexical scope. An example of this is registering channels with a `Selector`. How can we do this using `ZManaged` while still avoiding the possibility of leaks? One way is to use [the "scope" feature of `ZManaged`][zio-scope].
+
+A scope is itself a managed resource. Other managed resources can be attached to a scope, which gives them the same lifecycle as the scope. When the scope is released, all the other resources that have been attached to it will also be released.
+
+```scala mdoc:silent
+ZManaged.scope.use { scope =>
+
+  val channel: IO[IOException, SocketChannel] = scope(SocketChannel.open).map {
+    case (_ /* early release */, channel) => channel
+  }
+
+  // use channel, perhaps with a Selector
+  channel.flatMap(_.readChunk(10))
+
+}
+// the scope has now been released, as have all the resources attached to it
+```
+
+Note that `scope` returns both the resource and an "early release" effect. This allows you to release the resource before the scope exits, if you know it is no longer needed. This allows efficient use of the resource while still having the safety net of the scope to ensure the release happens even if there are failures, defects or interruptions.
+
+The `zio.nio.core.channels.SelectorSpec` test demonstrates the use of scoping to ensure nothing leaks if an error occurs.
+
+### Using `close` for Early Release
+
+In the case of channels, we don't actually need the early release features that `ZManaged` provides, as every channel has a built-in early release in the form of the `close` method. Closing a channel more than once is a perfectly safe thing to do, so you can use `close` to release a channel's resources early. When the `ZManaged` scope of the channel later ends, `close` will be called again, but it will be a no-op.
```

Comment:
Is it desirable to expose `close` to the user in addition to open/close as a `ZManaged`? With `ZManaged` you can be sure that the resource is still 'open' as long as the `ZManaged` is alive, but by exposing `close` that is no longer the case.

Reply:
That's true, but the same risk exists if you get the release effect via `withEarlyRelease` or `scope`.